### PR TITLE
Make xo implementations and tests conform to spec

### PR DESCRIPTION
**Pull request by nick-drozd**
The actual xo specification hasn't been written yet, so this PR expands the TP test to be spec-like and make the Python implementation clearer. It also fixes the xo state format to account for hash collisions.

Though it isn't strictly related to changes in the specification, the xo smoke test has been updated to actually verify results (previous, it just ran the xo cli commands and then passed if nothing blew up).

### CODE REVIEW

File: sdk/examples/xo_python/sawtooth_xo/xo_message_factory.py [hunk 1]
```diff
@@ -13,57 +13,65 @@
...
+
+def make_xo_address(name):
+    return XO_NAMESPACE + hashlib.sha512(name.encode('utf-8')).hexdigest()[-64:]
+
+# encodings
+def encode_txn_payload(action, name, space=''):
+    return ','.join([str(data) for data in (action, name, space)]).encode()
+
+
 class XoMessageFactory:
```
Comment (peterschwarz) [inline sdk/examples/xo_python/sawtooth_xo/xo_message_factory.py]: Could you add a commit to move this to a test directory - it shouldn't live with the production code.

File: sdk/examples/xo_javascript/xo_handler.js [hunk 2]
```diff
@@ -29,13 +29,21 @@ const _hash = (x) =>
 const XO_FAMILY = 'xo'
 const XO_NAMESPACE = _hash(XO_FAMILY).substring(0, 6)
 
+const VALID_ACTIONS = ['create', 'take']
+const VALID_STATES = ['P1-NEXT', 'P2-NEXT', 'P1-WIN', 'P2-WIN', 'TIE']
+
+const MARK_1 = 'X'
```
Comment (peterschwarz) [inline sdk/examples/xo_javascript/xo_handler.js]: Name these `MARK_X` and `MARK_O` - no need to obfuscate them.

File: sdk/examples/xo_python/sawtooth_xo/processor/handler.py [hunk 3]
```diff
@@ -21,16 +21,47 @@
...
+    return data.decode().split(',')
+
+def encode_state_data(data):
+    return ','.join(data).encode()
 
+# board
+SIDE_LENGTH = 3
+SPACE_COUNT = SIDE_LENGTH ** 2
+VALID_SPACES = range(1, SPACE_COUNT + 1)
+VALID_MARKS = MARK_1, MARK_2, EMPTY_SPACE = 'X', 'O', '-'
```
Comment (peterschwarz) [inline sdk/examples/xo_python/sawtooth_xo/processor/handler.py]: Name these `MARK_X` and `MARK_O`; no need for the obfuscation.

File: integration/sawtooth_integration/docker/two-families.yaml [hunk 4]
```diff
@@ -23,8 +23,6 @@ services:
       - $SAWTOOTH_CORE:/project/sawtooth-core
     expose:
       - 40000
-    depends_on:
```
Comment (boydjohnson) [inline integration/sawtooth_integration/docker/two-families.yaml]: These 'depends_on' deletions shouldn't be part of this PR. I mentioned in RocketChat that I had spread some bad information about needing these statements in the docker compose yaml files: https://chat.hyperledger.org/channel/sawtooth?msg=cM4rrDvj95WfaPrrD. We should verify that all of our integration tests can pass reliably without depends_on, and then make a PR that removes all of them.

File: sdk/examples/xo_python/sawtooth_xo/xo_message_factory.py (hunk 1 above)
Comment (nick-drozd) [inline sdk/examples/xo_python/sawtooth_xo/xo_message_factory.py]: I can do that in this PR, or open another PR to do it for all the TPs -- intkey and registry follow this same pattern, while config has a different but still odd arrangement.

File: sdk/examples/xo_python/sawtooth_xo/processor/handler.py (hunk 3 above)
Comment (nick-drozd) [inline sdk/examples/xo_python/sawtooth_xo/processor/handler.py]: Should they be renamed PLAYER_1_MARK, etc? They're the marks for players 1 and 2. They happen to be 'X' and 'O', but they could be different.

**Comment by nick-drozd** [comment]
@aludvik My hope was to get all the formal requirements into the TP test and then convert that into a document. The basic guiding principle is "do whatever the existing Python implementation does".

**Comment by nick-drozd** [comment]
Clearly there are some problems with the structure of this PR's commits, so I'm going to reorganize it.
